source/PlayerInfo.cpp

```diff
@@ -586,8 +586,19 @@ void PlayerInfo::AddChanges(list<DataNode> &changes)
 // Add an event that will happen at the given date.
 void PlayerInfo::AddEvent(const GameEvent &event, const Date &date)
 {
-	gameEvents.push_back(event);
-	gameEvents.back().SetDate(date);
+	// Check if the event should be applied directly.
+	if(date <= this->date)
+	{
+		GameEvent eventCopy = event;
+		list<DataNode> eventChanges = {eventCopy.Apply(*this)};
```

Comment:
Why copy the event? Btw there's nothing blocking GameEvent::Apply from being const?
```suggestion
		list<DataNode> eventChanges = {event.Apply(*this)};
```

Reply:
it is not const because of the std::move

Reply:
I copy it because it is not const

Reply:
ah, different std implementations in various compilers, I guess

Reply:
not really, https://en.cppreference.com/w/cpp/utility/move
if you look here you can see that std move does not take in const values

Reply:
std move empties the object it takes and "moves" the data into the other object

Reply:
~~so why did I manage to compile with const~~

Reply:
thats strange

Reply:
what compiler are you using?

Reply:
ah ok, no I get it now, so std move only tries to move stuff but when its const it just copies it

Reply:
hmm interesting

Reply:
then it can stay as is